packages/core/test/deterministic.test.ts

```diff
@@ -97,69 +97,56 @@ describe('deterministic machine', () => {
 
     it('should throw an error if not given an event', () => {
       expect(() =>
-        lightMachine.transition(
+        simulate(lightMachine).transition(
           testMachine.resolveStateValue('red'),
-          undefined as any,
-          undefined as any // TODO: figure out the simulation API
+          undefined as any
         )
       ).toThrow();
     });
 
     it('should transition to nested states as target', () => {
       expect(
-        testMachine.transition(
-          testMachine.resolveStateValue('a'),
-          { type: 'T' },
-          undefined as any // TODO: figure out the simulation API
-        ).value
+        simulate(testMachine).transition(testMachine.resolveStateValue('a'), {
```

Comment:
Do we have any ideas on how to put this kind of state resolution in the simulation API? Or do we just want to keep API and require people to resolve state values outside of the simulation API?

Reply:
Ideally `.resolveStateValue` is seldom used in real-world applications, in favor of real state. So, the latter is fine.